### ISSUE TITLE
"Stripe Payment Request live mode requires SSL" error despite site being provisioned with a SSL certificate

### ISSUE THREAD

**Issue by pauljacobson** [Automattic/woocommerce-payments#4247]
### Describe the bug

Our merchant reported that customers are unable to pay using Google Pay through WooCommerce Payments. Customers see the following error:

>There was an error processing your order

The site is a WordPress.com site with a domain that is provisioned with a SSL certificate using Let's Encrypt. The domain appears to be correctly configured. The site's System Status Report confirms that SSL is enabled too.

When we investigated further, we noticed that their WooCommerce Payments logs include the following:

```
2022-05-04T02:56:17+00:00 INFO Stripe Payment Request live mode requires SSL.
2022-05-04T02:56:19+00:00 INFO Stripe Payment Request live mode requires SSL.
2022-05-04T02:56:22+00:00 INFO Stripe Payment Request live mode requires SSL.
2022-05-04T02:56:22+00:00 INFO Stripe Payment Request live mode requires SSL.
2022-05-04T03:00:59+00:00 INFO REQUEST GET https://public-api.wordpress.com/wpcom/v2/sites/%s/wcpay/currency/rates?test_mode=0&currency_from=AUD
2022-05-04T03:00:59+00:00 INFO HEADERS: array (
  'Content-Type' => 'application/json; charset=utf-8',
  'User-Agent' => 'WooCommerce Payments/4.0.2',
)
```

This issue seems to be related to this issue affecting the Stripe gateway: [Site is using SSL but gets error in debug log. · Issue #1601 · woocommerce/woocommerce-gateway-stripe](https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1601).

### To Reproduce

1. Attempt to make a payment using the Google Pay payment option through WooCommerce Payments
2. See the error 

The merchant explained as follows:

>I set up Woocommerce Payments a few days ago and turned off Stripe but I wasn't getting sales and did some tests and found that when customers selected GPay on the product page or the cart page (fast checkout) it would say "There was an error processing your order".
>
>When going out of Gpay the error on website is- "The payment request button is not supported in Australia because some required fields couldn't be verified. Please proceed to the checkout page and try again. Billing street address is a required field. Billing Suburb is a required field. Billing State is a required field. (Customers have to be logged in to order on my website). This doesn't happen with paypal, when paypal is selected and the customer logs in to paypal, then it seems to go back to the website checkout and all address etc details are in the checkout and then they can submit the order. This doesn't seem to work for Gpay though, unsure if this happens to applepay as I don't have applepay account.
>
>I got my daughter to do a test on her computer and exactly the same thing happens. Can't pay using the woocommerce payment (Fast Checkout), I turned Stripe back on and the same thing happens with the fast checkout in product page and cart page. So I have disabled these until I find an answer.
>
>Gpay via Stripe seems to work from the checkout page only when all the checkout details have been entered but then when I select a card to pay the error: This card can't be verified right now. I select every card same error message pops up, I then enter the card number and details manually and it worked, the order went through. I am thinking customers are having the same issues.
>
>Also with woocommerce payments I put in test mode and used a test card number, it also wouldn't work in Gpay from product page but when I entered it in the checkout page the order went through.

We asked the user to verify that WooCommerce Payments is correctly configured, and has active webhooks.

### Actual behavior

Payments are intermittently failing when customers select Google Pay as a payment option.

### Screenshots

We don't seem to have screenshots currently.

### Expected behavior

Payments through Google Pay should be processed consistently where WooCommerce Payments is correctly configured.

### Desktop (please complete the following information):

* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

### Smartphone (please complete the following information):

* Device: [e.g. iPhone6]
* OS: [e.g. iOS8.1]
* Browser [e.g. stock browser, safari]
* Version [e.g. 22]

### Additional context

- The user also has Stripe installed on the site.
- They reported no change if they deactivate Stripe.
- Internal discussion here: p1651656519821979-slack-C7U3Y3VMY
- Ticket: 5174939-zen

**Comment by RadoslavGeorgiev** [comment]
We need to dig into this issue in order to understand why it is happening, as there is no apparent reason.

**Comment by zmaglica** [comment]
I did some digging and investigation upon this issue and I couldn't find the reason why this is happening. Also, I failed to recreate an issue on tests site. The reason for this issue can occur because of different reasons, and each of them is related to something different. After some time, the issue fix itself (AFAIK) and there are a couple of fixed that can be applied like checking WP site URL or is the website behind the proxy or something totally different.

To make this more resolvable, we should create a post about a few things that can be used to resolve this issue. If the error persists, it would be a good idea to take a look into configuration by the developer who will be more experienced with the debugging and digging into why this error occurs.
